### PR TITLE
Merge 3 components of Foreman Node Sharing into 1 repo

### PR DESCRIPTION
**Pull request by scoheb**

### CODE REVIEW

**Comment by scoheb** [comment]
Tests passing at https://ci.jenkins.io/job/Plugins/job/foreman-node-sharing-plugin/job/PR-51/1/

File: foreman-container/plugins/foreman_reserve.rb [hunk 1]
```diff
@@ -0,0 +1 @@
+gem 'foreman_reserve', :git => "https://github.com/david-caro/foreman_reserve.git"
```
Comment (olivergondza) [inline foreman-container/plugins/foreman_reserve.rb]: Is this using the beading-edge version of the plugin instead of a fixed verion?

**Comment by scoheb** [comment]
yes..."bleeding" edge -- see source code link at https://rubygems.org/gems/foreman_reserve/versions/0.1.8.3

Although no commit or release action in a while.

**Comment by olivergondza** [comment]
@pjanouse, can you have a look if this works for us? It looks good to me on a brief review.

**Comment by olivergondza** [comment]
Argh, merged accidentally. Reverting.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                timestamps {
+                    sh 'mvn -B -U -e -Dmaven.test.failure.ignore=true -Duser.home=/var/maven clean install -DskipTests'
+                }
+            }
+        }
+
+        stage('Test Plugin') {
+            docker.image('jenkins/ath').inside(containerArgs) {
+                sh '''
+                eval $(./vnc.sh 2> /dev/null)
```
Comment (olivergondza) [inline Jenkinsfile]: Do we really need ATH container with VNC running? I do not see anything browser specific ...

Comment (scoheb) [inline Jenkinsfile]: I think I was getting things ready to be able to run the Foreman ATH tests...

**Pull request merged**